### DISCUSSION TITLE
Natural Sort Order attribute

### DISCUSSION THREAD

**Discussion by nep** [ExpressionEngine/ExpressionEngine#3828] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
I'd like the ability to have "smart sorting" as a tag attribute for multiple situations:
* entries by field or title or url_title
* files
* categories
* members

By this, I mean I'd like to be able to return entries sorted as:
Entry 1
Entry 2
Entry 3
Entry 10
Entry 11
Entry 100

Instead of 1,10,11,100,2,3

**Describe the solution you'd like**
This is called Natural Sort Ordering and it would be great to have an attribute like `natural_sort='yes'` that could be used in all sorts of tags like the ones above.

https://en.wikipedia.org/wiki/Natural_sort_order

**Describe alternatives you've considered**
I've considered an add-on, but I think this would be better handled by modifying core.  Support? Thoughts?

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
I love the idea, though it might be not the easiest one to implement.

MariaDB has [`natural_sort_key`](https://mariadb.com/kb/en/natural_sort_key/) function, but MySQL does not have it. There are some workaround, but they are rather complex.

PHP does have `natsort` function, but when pagination exists applying it would become problematic
